### PR TITLE
Port firmware to Zephyr 1.6

### PR DESCRIPTION
**Pull request by bigdinotech**
-port firmware to be compatible with Zephyr 1.6
-remove soft-reset button since another functionality will be assigned
to it in the future

### CODE REVIEW

**Comment by bigdinotech** [comment]
@eriknyquist @SidLeung ready for review

File: src/cdcacm_serial.c [hunk 1]
```diff
@@ -171,7 +171,7 @@ void init_cdc_acm()
 	gpio_pin_write(gpio_dev, TXRX_LED, 1);
 }
 
-void cdcacm_setup(void)
+void cdcacm_setup(void *dummy1, void *dummy2, void *dummy3)
```
Comment (eriknyquist) [inline src/cdcacm_serial.c]: did you try `ARG_UNUSED(dummy1)`?

Comment (bigdinotech) [inline src/cdcacm_serial.c]: @eriknyquist I actually figured out the reason for the warnings. I forgot to change the function prototypes to have three arguments also. I'll amend the request later today

File: src/main.c [hunk 2]
```diff
@@ -0,0 +1,62 @@
+/*
```
Comment (calvinatintel) [inline src/main.c]: Why `main.c`, not `main.cpp`?

Comment (bigdinotech) [inline src/main.c]: .cpp would not compile right now with Zephyr 1.6 unless we change some of the Zephyr files

**Comment by bigdinotech** [comment]
@eriknyquist the warnings on firmware code has been fixed. There are still a couple warnings but they are from the cdc-acm driver that is using deperecated functions which is in Zephyr

**Comment by bigdinotech** [comment]
@eriknyquist hold off the review I will add the soft-reset functionality back as requested

File: src/main.c (hunk 2 above)
Comment (calvinatintel) [inline src/main.c]: Okay. Is `main.c` is a whole new file, not a derivation of the `main.cpp`? Just curious.

**Comment by bigdinotech** [comment]
@eriknyquist @SidLeung ready for review again. Soft-reset functionality has been ported also

File: src/cdcacm_serial.c (hunk 1 above)
Comment (eriknyquist) [inline src/cdcacm_serial.c]: So no warnings anymore, then? If there are no compile warnings I'm happy with not using ARG_UNUSED

Comment (SidLeung) [inline src/cdcacm_serial.c]: I suppose it is not a good idea to leave it open ended, cdcacm_setup()?

**Pull request merged**